client/src/dispatch.rs

```diff
@@ -42,6 +42,18 @@ where
             None => handler.dispatcher(from).await,
         };
 
+        if session.is_some() && packet.is_protocol_packet() {
+            let s = session.clone().unwrap();
+            if s.raw.id.to_vec() != packet.session_id() {
```

Comment:
Unimportant, but instead of cloning inside of `to_vec()` you could

```
impl Deref for SessionId {
    type Target = [u8; SESSION_ID_SIZE];

    fn deref(&self) -> &Self::Target {
        &self.id
    }
}
```

and then `if *s.raw.id != &*packet.session_id()`.